src/packagedcode/spec.py

```diff
@@ -0,0 +1,156 @@
+                stripped_emails = stripped_emails.strip()
+                stripped_emails = stripped_emails.split(',')
+                spec_data['email'] = stripped_emails
+            elif '.author' in line:
+                authors = re.sub(r'/*.*author.*?=', '', line)
+                stripped_authors = get_stripped_data(authors)
+                stripped_authors = stripped_authors.replace(' => ', "=>")
+                stripped_authors = stripped_authors.strip()
+                stripped_authors = stripped_authors.split(',')
+                spec_data['author'] = stripped_authors
+            elif '.summary' in line:
+                summary = re.sub(r'/*.*summary.*?=', '', line)
+                spec_data['summary'] = get_stripped_data(summary)
+            elif '.description' in line:
+                if location.endswith('.gemspec'):
+                    desc = re.sub(r'/*.*description.*?=', '', line)
+                    spec_data['description'] = get_stripped_data(desc)
+                else:
+                    spec_data['description'] = get_description(location)
+            elif '.homepage' in line:
+                homepage_url = re.sub(r'/*.*homepage.*?=', '', line)
+                spec_data['homepage_url'] = get_stripped_data(homepage_url)
+
+        parser = GemfileParser(location)
+        deps = parser.parse()
+        dependencies = OrderedDict()
+        for key in deps:
+            if deps[key]:
+                for dependency in deps[key]:
+                    dependencies[dependency.name] = (dependency.requirement).split(',')
+        spec_data['dependencies'] = dependencies
+
+        return spec_data
+
+
+def get_stripped_data(line):
+    """
+    Return data after removing unnecessary special character and space.
+    """
+    if '#' in line:
+        line = line[:line.index('#')]
+    stripped_data = line.strip()
+    stripped_data = stripped_data.replace("'", '')
+    stripped_data = stripped_data.replace('"', '')
+    stripped_data = stripped_data.replace('{', '')
+    stripped_data = stripped_data.replace('}', '')
+    stripped_data = stripped_data.replace('[', '')
+    stripped_data = stripped_data.replace(']', '')
+    stripped_data = stripped_data.replace('%q', '')
+    stripped_data = stripped_data.strip()
+    return stripped_data
+
+
+def get_description(location):
+    """
+    https://guides.cocoapods.org/syntax/podspec.html#description
+    description is in the form:
+    spec.description = <<-DESC
+                     Computes the meaning of life.
+                     Features:
+                     1. Is self aware
+                     ...
+                     42. Likes candies.
+                    DESC
+    Return description from podspec.
+    """
+    with io.open(location, encoding='utf-8', closefd=True) as data:
+        lines = data.readlines()
+    description = ''
+    for i, content in enumerate(lines):
+        if '.description' in content:
```

Comment:
What is there is no `<<-DESC` starting on that line?

Reply:
This function is for `.podspec` files not for `.gemspec` files. According to the documentation in `.podspec` files, description will be written in this manner but in gemspec file it is not fixed how to write a description.

Reply:
and what if this is using `<<-DESCRIPTION`? will this work too?

Reply:
I am following Cocoapods(https://guides.cocoapods.org/syntax/podspec.html#description) documentation. The general way is using `DESC` but using `DESC` will also work for `<<-DESCRIPTION` :)